server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -692,6 +753,18 @@ protected void doRun() {
                 }
             });
         }
+
+        // Unregister the closing channel and remove bans whose has no registered channels
+        final Iterator<Map.Entry<TaskId, Ban>> iterator = bannedParents.entrySet().iterator();
+        while (iterator.hasNext()) {
+            final Map.Entry<TaskId, Ban> entry = iterator.next();
+            final Ban ban = entry.getValue();
+            if (ban.perChannel) {
+                if (ban.unregisterChannel(channel) && entry.getValue().registeredChannels() == 0) {
+                    iterator.remove();
```

Comment:
This can happen. I've tightened the synchronization in https://github.com/elastic/elasticsearch/pull/66066/commits/76950a6c2acd98a70ced9f54bb0a867f3c80d6f8.